src/main/java/edu/umich/ctools/sectionsUtilityTool/Friend.java

```diff
@@ -0,0 +1,317 @@
+import java.security.SecureRandom;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.KeyFactory;
+
+import javax.net.ssl.*;
+import javax.net.SocketFactory;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.apache.xmlrpc.client.XmlRpcSun15HttpTransportFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+//TODO - Email Templates
+
+public class Friend 
+{
+
+	private static Log M_log = LogFactory.getLog(Friend.class);
+
+	//Currently it uses xmlrpc for communication
+	public static XmlRpcClient Xclient;
+
+	protected static String friendUrl = null;
+	protected static String contactEmail = null;
+	protected static String emailMessage = null;
+	protected static String referrerUrl = null;
+	protected static String ksFileName = null;
+	protected static String ksPwd = null;
+	protected static String friendEmailFile = null;
+	protected static String requesterEmailFile = null;
+
+	public final String KEYSTORETYPE_PKCS12 = "pkcs12";
+
+	public final String TRUSTSTORETYPE_JKS = "jks";
+
+	private static boolean sslInitialized = false;
+	
+	protected static final String FRIEND_PROPERTY_FILE_PATH_SECURE = "sectionsToolFriendPropsPathSecure";
+	
+	protected static Properties appExtSecurePropertiesFile=null;
+	
+
+	public void init()
+	{
+		M_log.info(this +".init()");
+		
+		String propertiesFilePathSecure = System.getProperty(FRIEND_PROPERTY_FILE_PATH_SECURE);
+		M_log.info("props: " + propertiesFilePathSecure);
+		if (!propertiesFilePathSecure.isEmpty()) {
+			appExtSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathSecure);
+			if(appExtSecurePropertiesFile!=null) {
+				//PropertiesFile information
+				friendUrl = appExtSecurePropertiesFile.getProperty("ctools.friend.url");
+				contactEmail = appExtSecurePropertiesFile.getProperty("ctools.friend.contactemail");
+				referrerUrl = appExtSecurePropertiesFile.getProperty("ctools.friend.referrer");
+				ksFileName = appExtSecurePropertiesFile.getProperty("ctools.friend.ksfilename");
+				ksPwd = appExtSecurePropertiesFile.getProperty("ctools.friend.kspassword");
+				friendEmailFile = appExtSecurePropertiesFile.getProperty("ctools.friend.friendemail");
+				requesterEmailFile = appExtSecurePropertiesFile.getProperty("ctools.friend.requesteremail");
+				
+				M_log.debug("friendUrl: " + friendUrl);
+				M_log.debug("contactEmail: " + contactEmail);
+				M_log.debug("referrerUrl: " + referrerUrl);
+				M_log.debug("ksFileName: " + ksFileName);
+				M_log.debug("ksPwd: " + ksPwd);
+			}else {
+				M_log.error("Failed to load secure application properties from sectionsToolPropsSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathSecure.properties) is not provided");
+		}
+
```

Comment:
I think in the error message property file name should be  `sectionsToolFriend.properties` instead of `sectionsToolPropsSecure.properties`.